notebooks/sentiment_analysis.py

```diff
@@ -0,0 +1,338 @@
+
+In this notebook we perform **sentiment analysis**: we use natural language processing models to classify text prompts. 
+We follow [this blog post by Federico Pascual](https://huggingface.co/blog/sentiment-analysis-python) and test 5 different models available on HuggingFace Hub to highlight different model properties of the models that can be leveraged for downstream tasks.
+
+The ease of use of the `pipeline` interface lets us quickly experiment with the pre-trained models and identify which one will work best.
+This simple interface means that it is extremely easy to access the fast inference performance of the IPU on your application.
+
+![Widget inference on a text classification task](images/text_classification.png)
+
+While this notebook is focused on using the model (inference), our [text_classification](text_classification.ipynb) notebook will show you how to fine tune a model for a specific task using the `datasets` package.
+"""
+"""
+First of all, lets make sure your environment has the latest version of [🤗 Optimum Graphcore](https://github.com/huggingface/optimum-graphcore) available.
+"""
+%pip install "optimum-graphcore>=0.4, <0.5"
+%pip install emoji == 0.6.0
+
+%load_ext autoreload
+%autoreload 2
+
+import os
+os.environ["POPTORCH_LOG_LEVEL"] = "ERR"
+
+"""
+## Using transformers pipelines on the IPU
+
+The simplest way to use a model on the IPU is to use the `pipeline` function. It provides a set of models which have been validated to work on a given task, to get started choose the task and call the `pipeline` function:
+"""
+from optimum.graphcore import pipelines
+sentiment_pipeline = pipelines.pipeline("sentiment-analysis")
+
+"""
+Because no model was supplied the pipeline defaulted to `distilbert-base-uncased-finetuned-sst-2-english`. We are helpfully warned that we should provide the `max_length` argument if we were to put this model in production, but while we are experimenting we will leave it as is.
+
+Now it's time to test our first prompts. Let's start with some very easy to classify text:
+"""
+simple_test_data = ["I love you", "I hate you"]
+sentiment_pipeline(simple_test_data)
+
+"""
+Reassuringly, the model got it right! And with a high degree of confidence, more than 99.9% in both cases.
+
+You'll have noticed that this was a bit slow, it took several seconds to provide the answer. This is because of the computing architecture of the IPU, during the first execution of a pipeline, the model needs to be compiled and loaded into the IPU. While this takes some time, these overheads only the first execution of a pipeline! 
+
+On subsequent prompts it is much faster:
+"""
+sentiment_pipeline(simple_test_data)
+
+"""
+Now that was much faster! We can use the `%%timeit` cell magic to check how fast:
+"""
+% % timeit
+sentiment_pipeline(simple_test_data)
+
+"""
+It takes on the order of ~1ms per prompt, this is fast!
+"""
+"""
+### Customising the IPU resources
```

Comment:
I'm not sure this section is needed.

Reply:
Well I hate it, but I need the `inference_config` to be set otherwise nothing works the alternative is to change defaults in the library. Which I'd like to do after the release but seems risky before. 

Given that I prefer addressing the arguments front and centre, I feel it is beneficial to drip feed a bit of IPU information in the tutorial so that if they then read tutorials that are more IPU specific it's not a total surprise that the integration is not perfect.